Don't retry skipped tests.

jmchilton opened the pull request:
Wasting time on these when running Selenium tests on Jenkins.

jmchilton committed:
Don't retry skipped tests.

Wasting time on these when running Selenium tests on Jenkins.

nsoranzo commented on test/selenium_tests/framework.py:
```diff
@@ -145,6 +145,10 @@ def func_wrapper(self, *args, **kwds):
+                raise
             except Exception:
                 dump_test_information(self, f.__name__)
                 if retry_attempts < GALAXY_TEST_SELENIUM_RETRIES:
```
It was a bit hard for me to find in the console log that a failed test was retried (e.g. both JupyterTestCase tests in https://jenkins.galaxyproject.org/job/docker-selenium/1716/console), the only clue for me was that `Binding default provider to: org.openqa.selenium.chrome.ChromeDriverService` was repeated. Can we insert a log statement after this line?

jmchilton committed:
Print a statement when retrying Selenium test functions.

Pointed out by @nsoranzo (https://github.com/galaxyproject/galaxy/pull/5708#discussion_r175054796) that there isn't mmuch of an indication this is occurring otherwise.

nsoranzo commented:
I was monitoring the selenium test logs in https://jenkins.galaxyproject.org/job/docker-selenium/1727/ and I noticed that the tests in `test/selenium_tests/test_jupyter.py` were retried any way, I suppose because they are annotated with:
```
    @flakey
    @selenium_test
```
instead of:
```
    @selenium_test
    @flakey
```

Same applies for `test_run_data` in `test/selenium_tests/test_tool_form.py` .
